rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/BookKeeper.java

```diff
@@ -296,6 +297,43 @@ public void onRemoval(
                 .build();
     }
 
+    @Override
+   public boolean readData(String path, long offset, int length, long fileSize, long lastModified, int clusterType)
+           throws TException
+   {
+       int blockSize = CacheConfig.getBlockSize(conf);
+       byte[] buffer = new byte[blockSize];
```

Comment:
Take this buffer out as a static variable of this class.
The data written to it is not used at all and we will waste GC cycles because of so many allocations of this buffer. Better to keep one copy and let all use it to dump data. Add a warning at the declaration of this that data in this buffer will always be garbage due to parallel writes and should not be used anytime